### PR TITLE
Feature/json nullwrite

### PR DESCRIPTION
**Pull request by bengalengel**
na="null" passed to write_json for the export of model metadata.

### CODE REVIEW

File: R/utility.R [hunk 1]
```diff
@@ -65,7 +65,13 @@ readJson <- function(x, simplifyVector = TRUE, ...) {
 }
 
 writeJson <- function(x, file, auto_unbox = TRUE, pretty = TRUE, ...) {
-  jsonlite::write_json(x, path = file, auto_unbox = auto_unbox, pretty = pretty, ...)
+  printingArgs <- list(...)
```
Comment (jdblischak) [inline R/utility.R:68]: This part shouldn't be needed if `na="null"` is propagated via `...` through all the function calls

Comment (bengalengel) [reply R/utility.R:68]: which part? The `list(...)` call? If so, I've tried `na=..1` directly in the `jsonlite::writejson` call and that works, too. The purpose of the list call is to accommodate multiple arguments if they are eventually passed down in a future update. 

Also, I'm using the `...` here so I can be free to modify the upstream `export` calls with additional arguments to `jsonlite::write_json`. In this change, I am only using an addiitonal argument upstream with `exportModels`, hence the ifelse logic. The other `export` functions don't have the `na="null"` argument.

Comment (jdblischak) [reply R/utility.R:68]: The whole thing  seems unnecessary to me. `writeJson()` already passes additional arguments to `jsonlite::write_args()` via `...`. I don't see the advantage of capturing the arguments in a list and then manually adding them. Can you re-try with the original version of `writeJson()` and see if `na = "null"` is passed correctly? You can even do `debug(jsonlite::write_json)` to enter the interactive debugger and print the value of `na`

Comment (bengalengel) [reply R/utility.R:68]: The argument is being passed down from `exportModels()` not from `writeJson()`. The chain is like this:

> 	exportModels
       exportElements (self-referencing function)
						exportElementsWrite (non-nested base case)
							writeJson(…)
								jsonlite::write_json(…)

But yeah, I was hoping I wouldn't have to reference `...` at all and `jsonlite::write_json` would just recognize `na=null`. This was what I checked first without the iflese logic. Didn't work but I'll check again to be sure and get back to you.

Comment (bengalengel) [reply R/utility.R:68]: yeah, it works with the `...` only... Thanks for the review John!

**Comment by bengalengel** [comment]
@jdblischak if you're good with this I'd like to squash and merge this one.

File: NEWS.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
+* export of metadata to JSON now follows the convention NA -> null
```
Comment (jdblischak) [inline NEWS.md:1]: This makes it sound like it applies to all JSON output

File: R/export.R [hunk 3]
```diff
@@ -70,7 +70,7 @@ createTextFiles <- function(study, directoryname, calcOverlaps = FALSE) {
   dir.create(directoryname, showWarnings = FALSE, recursive = TRUE)
   exportSamples(study, directoryname)
   exportFeatures(study, directoryname)
-  exportModels(study, directoryname)
+  exportModels(study, directoryname, na="null")
```
Comment (jdblischak) [inline R/export.R:73]: What is the motivation for only wanting to apply this to the "models" output and not other JSON output like "tests" and "annotations"?

File: R/utility.R [hunk 4]
```diff
@@ -65,7 +65,7 @@ readJson <- function(x, simplifyVector = TRUE, ...) {
 }
 
 writeJson <- function(x, file, auto_unbox = TRUE, pretty = TRUE, ...) {
```
Comment (jdblischak) [inline R/utility.R:67]: If you wanted to apply `na = "null"` to all JSON output files, you could treat it the same as other default arguments like `auto_unbox` and `pretty`. Add `na = "null"` to `writeJson()` and `na = na` to `jsonlite::write_json()`

**Review by jdblischak — APPROVED** [review]
I'm fine if you squash and merge this PR. However, I left some comments and suggestions for applying `na = "null"` globally for all JSON output if that would be desired

File: R/utility.R (hunk 4 above)
Comment (bengalengel) [reply R/utility.R:67]: yes

File: R/export.R (hunk 3 above)
Comment (bengalengel) [reply R/export.R:73]: Companion tools that need to be updated in sequence. This gives us flexibility with output types, which I need (for now) internally. This change won't impact any external users of OmicNavigator. I anticipate `na=null` to become the default argument for `write_json` in time.

File: NEWS.md (hunk 2 above)
Comment (bengalengel) [reply NEWS.md:1]: yes. will specify this then squash and merge

**Pull request merged**